### ISSUE TITLE
No gradient is provided to any variable. This occurs only when I want to use lambda layer for keras.argmax function and keep a model non trainable

### ISSUE THREAD

**Issue by Mizanur4E** [tensorflow/tensorflow#52260]
`def define_gan(g_model, d_model,timestep):
  
    discriminator.trainable = False
    generator.trainable= True
    
    a = Input(shape=(timestep,))
    b = g_model(a)
    c=K.argmax(b,axis=-1)
    output_d = d_model(c)
    model = Model(inputs=a,outputs=output_d)
    
    # compile model
    opt = Adam(lr=0.0002, beta_1=0.5)
    model.compile(loss=['binary_crossentropy'], optimizer=opt,metrics=['accuracy'])
    model.summary()
    return model`

**Comment by Mizanur4E** [comment]
Also When I keep discriminator.trainable= True. The gan model works fine and learns.

**Comment by Mizanur4E** [comment]
Interestingly if the generator is kept non-trainable and the discriminator keeps trainable then also it works. Only for the case when generator trainable and discriminator non-trainable produce the issue. In that case, is it the lamda layer causing the issue? Generator learns outside the gan model properly.


![image](https://user-images.githubusercontent.com/66197947/136146755-afffd33c-696e-44b0-9ea2-f8508bc5d1d9.png)

**Comment by Mizanur4E** [comment]
https://gist.github.com/Mizanur4E/0223074ea9f45b33b42357155b3a5b46
please have a look

**Comment by Mizanur4E** [comment]
@mohantym  have found the solution?
